controller/zuriInternController.js

```diff
@@ -1,16 +1,45 @@
 const ZuriIntern = require('../models/ZuriInternModel');
 const {
-  responseHandler
+    responseHandler
 } = require('../utils/responseHandler');
 
 module.exports = {
-  getAllInterns: async (req, res) => {
-    try {
-      const zuriInterns = await ZuriIntern.find();
-      return responseHandler(res, 'Success', 200, true, zuriInterns);
-    } catch (err) {
-      return responseHandler(res, 'Error', 500, false, err);
+    getAllInterns: async (req, res) => {
+        let searchValue;
+        if (req.query.firstName) {
+            searchValue = {
+                firstName: req.query.firstName
```

Comment:
extend this search to check both first name and last name and please use regex if possible. it does not have to be a perfect match. 
Please reference what I did with Internship GET mentors